[BUG] (Impossible compilation)

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Trying to compile 2.0.8 (bugfix is idle)

Error message (partial)

Compiling .pio\build\LPC1769\src\src\lcd\menu\menu_tmc.cpp.o
In file included from Marlin\src\lcd\menu\menu_temperature.cpp:31:
Marlin\src\lcd\menu\menu_temperature.cpp: In function 'void menu_temperature()':
Marlin\src\lcd\menu\menu_temperature.cpp:174:57: error: invalid conversion from 'celsius_t*' {aka 'short int*'} to 'TMenuEditItem<MenuEditItemInfo_uint16_3>::type_t*' {aka 'short unsigned int*'} [-fpermissive]  174 |       EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                         |
      |                                                         celsius_t* {aka short int*}
Marlin\src\lcd\menu\menu_item.h:261:39: note: in definition of macro '_MENU_INNER_P'
  261 |     MenuItem_##TYPE::action(plabel, ##V);                      \
      |                                       ^
Marlin\src\lcd\menu\menu_item.h:402:59: note: in expansion of macro '_MENU_ITEM_N_P'
  402 | #define EDIT_ITEM_FAST_N_P(TYPE, N, PLABEL, V...)         _MENU_ITEM_N_P(TYPE, N, true, PLABEL, ##V)
      |                                                           ^~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_item.h:403:55: note: in expansion of macro 'EDIT_ITEM_FAST_N_P'
  403 | #define EDIT_ITEM_FAST_N(TYPE, N, LABEL, V...)        EDIT_ITEM_FAST_N_P(TYPE, N, GET_TEXT(LABEL), ##V)
      |                                                       ^~~~~~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_temperature.cpp:174:7: note: in expansion of macro 'EDIT_ITEM_FAST_N'
  174 |       EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
      |       ^~~~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_item.h:95:22: note:   initializing argument 2 of 'static void TMenuEditItem<NAME>::action(const char*, TMenuEditItem<NAME>::type_t*, TMenuEditItem<NAME>::type_t, TMenuEditItem<NAME>::type_t, screenFunc_t, bool) [with NAME = MenuEditItemInfo_uint16_3; TMenuEditItem<NAME>::type_t = short unsigned int; screenFunc_t = void (*)()]'
   95 |       type_t * const ptr,                   // Value pointer
      |       ~~~~~~~~~~~~~~~^~~
Marlin\src\lcd\menu\menu_temperature.cpp:174:57: error: invalid conversion from 'celsius_t*' {aka 'short int*'} to 'TMenuEditItem<MenuEditItemInfo_uint16_3>::type_t*' {aka 'short unsigned int*'} [-fpermissive]  174 |       EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                         |
      |                                                         celsius_t* {aka short int*}
Marlin\src\lcd\menu\menu_item.h:266:58: note: in definition of macro '_MENU_INNER_P'
  266 |       (encoderLine == _thisItemNr, _lcdLineNr, plabel, ##V);   \
      |                                                          ^
Marlin\src\lcd\menu\menu_item.h:402:59: note: in expansion of macro '_MENU_ITEM_N_P'
  402 | #define EDIT_ITEM_FAST_N_P(TYPE, N, PLABEL, V...)         _MENU_ITEM_N_P(TYPE, N, true, PLABEL, ##V)
      |                                                           ^~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_item.h:403:55: note: in expansion of macro 'EDIT_ITEM_FAST_N_P'
      |                                                       ^~~~~~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_temperature.cpp:174:7: note: in expansion of macro 'EDIT_ITEM_FAST_N'
  174 |       EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
      |       ^~~~~~~~~~~~~~~~
Marlin\src\lcd\menu\menu_item.h:85:103: note:   initializing argument 4 of 'static void TMenuEditItem<NAME>::draw(bool, uint8_t, const char*, TMenuEditItem<NAME>::type_t*, ...) [with NAME = MenuEditItemInfo_uint16_3; uint8_t = unsigned char; TMenuEditItem<NAME>::type_t = short unsigned int]'
   85 |     FORCE_INLINE static void draw(const bool sel, const uint8_t row, PGM_P const pstr, type_t * const data, ...) {
      |                                                                                        ~~~~~~~~~~~~~~~^~~~
*** [.pio\build\LPC1769\src\src\lcd\menu\menu_temperature.cpp.o] Error 1

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/6408883/Marlin.zip)


### Bug Timeline

New

### Expected behavior

Compilation success!

### Actual behavior

Compilation issue

### Steps to Reproduce

Compile 2.0.8 with my config files

### Version of Marlin Firmware

2.0.8

### Printer model

FLSUN Prusa I3+

### Electronics

BTT SKR V1.4 TURBO

### Add-ons

_No response_

### Your Slicer

_No response_

### Host Software

SD Card (headless)

### Additional information & file uploads

_No response_

--- a/Marlin/src/inc/Conditionals_LCD.h
+++ b/Marlin/src/inc/Conditionals_LCD.h
@@ -641,6 +641,16 @@
 #endif
 
 /**
+ * Disable unused SINGLENOZZLE sub-options
+ */
+#if DISABLED(SINGLENOZZLE)
+  #undef SINGLENOZZLE_STANDBY_TEMP
+#endif
+#if !BOTH(HAS_FAN, SINGLENOZZLE)
+  #undef SINGLENOZZLE_STANDBY_FAN
+#endif
+
+/**
  * DISTINCT_E_FACTORS affects how some E factors are accessed
  */
 #if ENABLED(DISTINCT_E_FACTORS) && E_STEPPERS > 1

--- a/Marlin/src/lcd/menu/menu_temperature.cpp
+++ b/Marlin/src/lcd/menu/menu_temperature.cpp
@@ -171,7 +171,7 @@ void menu_temperature() {
 
   #if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
     LOOP_S_L_N(e, 1, EXTRUDERS)
-      EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
+      EDIT_ITEM_FAST_N(int3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
   #endif
 
   //

--- a/Marlin/src/lcd/menu/menu_tune.cpp
+++ b/Marlin/src/lcd/menu/menu_tune.cpp
@@ -134,7 +134,7 @@ void menu_tune() {
 
   #if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
     LOOP_S_L_N(e, 1, EXTRUDERS)
-      EDIT_ITEM_FAST_N(uint16_3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
+      EDIT_ITEM_FAST_N(int3, e, MSG_NOZZLE_STANDBY, &thermalManager.singlenozzle_temp[e], 0, thermalManager.hotend_max_target(0));
   #endif
 
   //

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -464,7 +464,7 @@ volatile bool Temperature::raw_temps_ready = false;
           Temperature::soft_pwm_count_fan[FAN_COUNT];
 #endif
 
-#if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
+#if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
   celsius_t Temperature::singlenozzle_temp[EXTRUDERS];
   #if HAS_FAN
     uint8_t Temperature::singlenozzle_fan_speed[EXTRUDERS];

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -464,11 +464,11 @@ volatile bool Temperature::raw_temps_ready = false;
           Temperature::soft_pwm_count_fan[FAN_COUNT];
 #endif
 
-#if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
+#if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
   celsius_t Temperature::singlenozzle_temp[EXTRUDERS];
-  #if HAS_FAN
-    uint8_t Temperature::singlenozzle_fan_speed[EXTRUDERS];
-  #endif
+#endif
+#if ENABLED(SINGLENOZZLE_STANDBY_FAN)
+  uint8_t Temperature::singlenozzle_fan_speed[EXTRUDERS];
 #endif
 
 #if ENABLED(PROBING_HEATERS_OFF)
@@ -2500,20 +2500,22 @@ void Temperature::disable_all_heaters() {
 
 #endif // PROBING_HEATERS_OFF
 
-#if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
+#if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
 
   void Temperature::singlenozzle_change(const uint8_t old_tool, const uint8_t new_tool) {
-    #if HAS_FAN
+    #if ENABLED(SINGLENOZZLE_STANDBY_FAN)
       singlenozzle_fan_speed[old_tool] = fan_speed[0];
       fan_speed[0] = singlenozzle_fan_speed[new_tool];
     #endif
-    singlenozzle_temp[old_tool] = temp_hotend[0].target;
-    if (singlenozzle_temp[new_tool] && singlenozzle_temp[new_tool] != singlenozzle_temp[old_tool]) {
-      setTargetHotend(singlenozzle_temp[new_tool], 0);
-      TERN_(AUTOTEMP, planner.autotemp_update());
-      TERN_(HAS_STATUS_MESSAGE, set_heating_message(0));
-      (void)wait_for_hotend(0, false);  // Wait for heating or cooling
-    }
+    #if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
+      singlenozzle_temp[old_tool] = temp_hotend[0].target;
+      if (singlenozzle_temp[new_tool] && singlenozzle_temp[new_tool] != singlenozzle_temp[old_tool]) {
+        setTargetHotend(singlenozzle_temp[new_tool], 0);
+        TERN_(AUTOTEMP, planner.autotemp_update());
+        TERN_(HAS_STATUS_MESSAGE, set_heating_message(0));
+        (void)wait_for_hotend(0, false);  // Wait for heating or cooling
+      }
+    #endif
   }
 
 #endif

--- a/Marlin/src/module/temperature.h
+++ b/Marlin/src/module/temperature.h
@@ -373,8 +373,10 @@ class Temperature {
     static inline bool targetHotEnoughToExtrude(const uint8_t e) { return !targetTooColdToExtrude(e); }
 
     #if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
-      static celsius_t singlenozzle_temp[EXTRUDERS];
-      #if HAS_FAN
+      #if ENABLED(SINGLENOZZLE_STANDBY_TEMP)
+        static celsius_t singlenozzle_temp[EXTRUDERS];
+      #endif
+      #if ENABLED(SINGLENOZZLE_STANDBY_FAN)
         static uint8_t singlenozzle_fan_speed[EXTRUDERS];
       #endif
       static void singlenozzle_change(const uint8_t old_tool, const uint8_t new_tool);

--- a/Marlin/src/module/temperature.h
+++ b/Marlin/src/module/temperature.h
@@ -372,7 +372,7 @@ class Temperature {
     static inline bool hotEnoughToExtrude(const uint8_t e) { return !tooColdToExtrude(e); }
     static inline bool targetHotEnoughToExtrude(const uint8_t e) { return !targetTooColdToExtrude(e); }
 
-    #if ENABLED(SINGLENOZZLE_STANDBY_FAN)
+    #if EITHER(SINGLENOZZLE_STANDBY_TEMP,SINGLENOZZLE_STANDBY_FAN)
       static celsius_t singlenozzle_temp[EXTRUDERS];
       #if HAS_FAN
         static uint8_t singlenozzle_fan_speed[EXTRUDERS];

--- a/Marlin/src/module/temperature.h
+++ b/Marlin/src/module/temperature.h
@@ -372,7 +372,7 @@ class Temperature {
     static inline bool hotEnoughToExtrude(const uint8_t e) { return !tooColdToExtrude(e); }
     static inline bool targetHotEnoughToExtrude(const uint8_t e) { return !targetTooColdToExtrude(e); }
 
-    #if EITHER(SINGLENOZZLE_STANDBY_TEMP,SINGLENOZZLE_STANDBY_FAN)
+    #if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
       static celsius_t singlenozzle_temp[EXTRUDERS];
       #if HAS_FAN
         static uint8_t singlenozzle_fan_speed[EXTRUDERS];

--- a/Marlin/src/module/tool_change.cpp
+++ b/Marlin/src/module/tool_change.cpp
@@ -1194,7 +1194,9 @@ void tool_change(const uint8_t new_tool, bool no_move/*=false*/) {
       const bool should_move = safe_to_move && !no_move && IsRunning();
       if (should_move) {
 
-        TERN_(SINGLENOZZLE_STANDBY_TEMP, thermalManager.singlenozzle_change(old_tool, new_tool));
+        #if EITHER(SINGLENOZZLE_STANDBY_TEMP, SINGLENOZZLE_STANDBY_FAN)
+          thermalManager.singlenozzle_change(old_tool, new_tool);
+        #endif
 
         #if ENABLED(TOOLCHANGE_FILAMENT_SWAP)
           if (should_swap && !too_cold) {

